### ISSUE TITLE
Graphviz with Asciidoctor: "double quotted string"

### ISSUE THREAD

**Issue by GSebastian-Adil** [asciidoctor/asciidoctor#3140]
I use graphviz with asciidoctor, but I can't set double quoted string as ID for digraph:
[Example_Diagram.txt](https://github.com/asciidoctor/asciidoctor/files/2949803/Example_Diagram.txt)
1) change the extension to (.adoc)
2) open it with firefox
=> an error message for the second diagram

Thanks

**Comment by Mogztter** [comment]
Graphviz support is provided by http://plantuml.com
I get the same error if I type the following diagram in the textarea:

```
digraph "Reflux Gastro Oesophagien" {
   "Reflux Gastro \n Oesophagien" -> "Oesophagite \n Peptique" -> "Stenose \n Peptique"
}
```

I'm not familiar with this syntax, why do you need to set an ID ?

**Comment by Mogztter** [comment]
I can confirm that this is a limitation of the PlantUML server. You can open a new issue here: https://github.com/plantuml/plantuml

**Comment by mojavelinux** [comment]
I see no error if used with the Ruby version of Asciidoctor and Asciidoctor Diagram. So this is not an issue with Asciidoctor core.

**Comment by GSebastian-Adil** [comment]
1. -Is people of graphviz how had sended me to you [https://gitlab.com/graphviz/graphviz/issues/1507](url).
2. -And when I use "asciidoctor-pdf Example_Diagram.adoc" command I get the message:
"asciidoctor: WARNING: Example_Diagram.adoc: line 5: invalid style for literal block: graphviz"
"asciidoctor: WARNING: Example_Diagram.adoc: line 13: invalid style for literal block: graphviz"
3. -What "PlantUml" has to do with Graphviz?

Thanks

**Comment by Mogztter** [comment]
> Is people of graphviz how had sended me to you https://gitlab.com/graphviz/graphviz/issues/1507.

Stephen C. North is correct, your diagram is correctly processed using GraphViz.

> And when I use "asciidoctor-pdf Example_Diagram.adoc" command I get the message:
"asciidoctor: WARNING: Example_Diagram.adoc: line 5: invalid style for literal block: graphviz"
"asciidoctor: WARNING: Example_Diagram.adoc: line 13: invalid style for literal block: graphviz"

I don't think it's related.
If I had to guess, I would say that Asciidoctor Diagram is not properly loaded, that's why asciidoctor-pdf does not recognize the "graphviz" block.

> What "PlantUml" has to do with Graphviz?

Behind the scene, the browser extension is using https://github.com/eshepelyuk/asciidoctor-plantuml.js
This extension mimics the behavior of the PlantUML support in Asciidoctor Diagram.
To do that the extension will use a PlantUML Server (by default it will use a public instance: http://www.plantuml.com/plantuml).

In other words your diagram is processed by the PlantUML server (not directly by GraphViz).
Since PlantUML is using GraphViz to render diagrams, it is possible to directly use DOT language: http://plantuml.com/dot

But unfortunately the syntax `digraph "XXXX" {` is not supported. So either the PlantUML server or the PlantUML core library does not recognize this syntax. So you should open a new issue in one of the following repositories:

- https://github.com/plantuml/plantuml
- https://github.com/plantuml/plantuml-server
